go/engine/login_with_paperkey.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+// This engine makes sure the user is logged in and unlocked.
+// It asks for a paper key if need be. It does not ask for a passphrase.
+
+package engine
+
+import "github.com/keybase/client/go/libkb"
+
+// LoginWithPaperKey is an engine.
+type LoginWithPaperKey struct {
+	libkb.Contextified
+}
+
+// NewLoginWithPaperKey creates a LoginWithPaperKey engine.
+// Uses the paperkey to log in and unlock LKS.
+func NewLoginWithPaperKey(g *libkb.GlobalContext) *LoginWithPaperKey {
+	return &LoginWithPaperKey{
+		Contextified: libkb.NewContextified(g),
+	}
+}
+
+// Name is the unique engine name.
+func (e *LoginWithPaperKey) Name() string {
+	return "LoginWithPaperKey"
+}
+
+// GetPrereqs returns the engine prereqs.
+func (e *LoginWithPaperKey) Prereqs() Prereqs {
+	return Prereqs{}
+}
+
+// RequiredUIs returns the required UIs.
+func (e *LoginWithPaperKey) RequiredUIs() []libkb.UIKind {
+	return []libkb.UIKind{
+		libkb.LogUIKind,
```

Comment:
Yeah this caused problems in `passphrase_change` engine but not this engine. I don't know why. All works now though.